File: src/main/java/de/kevcodez/metronom/model/stop/Station.java
package de.kevcodez.metronom.model.stop;

/**
 * Represents a single station.
 * 
 * @author Kevin Grüneberg
 *
 */
public class Station {

  private String name;
  private String code;

  public Station(String name, String code) {
    this.name = name;
    this.code = code;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getCode() {
    return code;
  }

  public void setCode(String code) {
    this.code = code;
  }

}

File: src/main/kotlin/de/kevcodez/metronom/model/station/Station.kt
package de.kevcodez.metronom.model.station

import com.fasterxml.jackson.annotation.JsonIgnore
import java.util.*

/**
 * Represents a single station. The station also contains a list of alternative names to resolve the station more
 * easily. For example the station Hannover Hbf also has the alternative name Hannover.
 *
 * @author Kevin Grüneberg
 */
data class Station(
    val name: String,
    val code: String
) {

    @JsonIgnore
    val alternativeNames: MutableList<String> = ArrayList()

    fun addAlternativeNames(vararg names: String) {
        alternativeNames.addAll(Arrays.asList(*names))
    }

}
